### DISCUSSION TITLE
the weak form in kobayashi module

### DISCUSSION THREAD

**Discussion by CGH20171006** [idaholab/moose#21676] [Q&A Modules: Phase field] [answered]
I can not correctly derive the weak form in the kobayshi model.The process is as follows:
![23A243D89E807461772DE3D5CFEF5D45](https://user-images.githubusercontent.com/107262618/180761046-f2c0495b-4e75-44f6-92c7-6c68a0c1104a.png)

The weak form in the READEME as follows:
![image](https://user-images.githubusercontent.com/107262618/180761295-deccb9c2-5a2d-4614-97b5-664ed731b9e5.png)

**Comment by CGH20171006** [comment, marked as answer]
and could you tell me why it is not necessary to define eps and deps in the \projects\moose\modules\phase_field\examples\anisotropic_interfaces\ad_snow.i ？
![image](https://user-images.githubusercontent.com/107262618/180771367-f7d26650-493a-44bd-a53b-2d15c3d95668.png)
![image](https://user-images.githubusercontent.com/107262618/180771419-8c182f28-640e-4140-90d5-2d168165dc13.png)

**Reply by CGH20171006** [reply]
Let me rephrase the question, how do I modify the formulas for eps and deps in that example (e.g. change the isotropic factor from 4 to 2)

**Reply by GiudGiud** [reply]
eps and deps are material properties, so you may modify them there.
https://mooseframework.inl.gov/syntax/Materials/
If you just want to add a factor you could just use an ADParsedMaterial for that
https://mooseframework.inl.gov/source/materials/ParsedMaterial.html

**Reply by CGH20171006** [reply]
> eps and deps are material properties, so you may modify them there. https://mooseframework.inl.gov/syntax/Materials/ If you just want to add a factor you could just use an ADParsedMaterial for that https://mooseframework.inl.gov/source/materials/ParsedMaterial.html OK,I got you,I find that in materials/ADInterfaceOrientationMaterial.C,thank you so much!

**Comment by CGH20171006** [comment]
> I can not correctly derive the weak form in the kobayshi model.The process is as follows: ![23A243D89E807461772DE3D5CFEF5D45](https://user-images.githubusercontent.com/107262618/180761046-f2c0495b-4e75-44f6-92c7-6c68a0c1104a.png)
> 
> The weak form in the READEME as follows: ![image](https://user-images.githubusercontent.com/107262618/180761295-deccb9c2-5a2d-4614-97b5-664ed731b9e5.png)

I still do not know how to derive this weak form : (
